packages/react-composites/src/composites/CallComposite/components/MediaGallery.tsx

```diff
@@ -134,24 +135,22 @@ export const useLocalVideoStartTrigger = (isLocalVideoAvailable: boolean, should
  * Hook to fetch new participant avatar and video gallery displayName information if it is present.
  * @param onFetchAvatarPersonaData
  */
-const useOnFetchAvatarPersonaData = (
+const useRemoteParticipantsWithCustomDisplayNames = (
   remoteParticipants: VideoGalleryRemoteParticipant[],
   onFetchAvatarPersonaData?: AvatarPersonaDataCallback
 ): VideoGalleryRemoteParticipant[] => {
-  // Sort the remote participants so that they are always in order
-  remoteParticipants.sort((a, b) => (a.userId < b.userId ? 1 : -1));
-
+  const newParticipants = remoteParticipants;
```

Comment:
Assigning `remoteParticipants` doesn't copy the items in the list.

To ensure that you don't update arguments, you'd have to update the logic on line 148 to create a copy of each participatn. Something like 
```
remoteParticipants.map((participant, i) => {
   const newName = avatarPersonaData[i]?.text;
   if (!newName) {
      return participant;
   };
   return {...participant, displayName: newName}; 
});
```